connection.go

```diff
@@ -567,6 +567,7 @@ func (sc *snowflakeConn) WaitForQueryCompletion(ctx context.Context, qid string)
 type ResultFetcher interface {
 	FetchResult(ctx context.Context, qid string) (driver.Rows, error)
 	WaitForQueryCompletion(ctx context.Context, qid string) error
+	SubmitAsync(context.Context, string, []driver.NamedValue) (queryID, isComplete, error) // leave this in for rebase
```

Comment:
Why is `ResultFetcher` doing submit ?

Reply:
I should create a separate one for wait and submit. I will do that

Reply:
done